inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/Initializer.kt

```diff
@@ -93,4 +98,39 @@ internal object Initializer {
         }
         Timber.tag(TAG).d(Gson().toJson(hostAppInfo))
     }
+
+    /**
+     * This method retrieves the stored UUID or generates a random ID if not available.
+     * This value is only used if Settings.Secure.ANDROID_ID returns a null value.
+     */
+    @SuppressWarnings("LongMethod")
+    private fun getUuid(context: Context): String {
+        val master = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val spec = KeyGenParameterSpec.Builder(
+                    MasterKey.DEFAULT_MASTER_KEY_ALIAS,
+                    KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT)
+                    .setBlockModes(KeyProperties.BLOCK_MODE_GCM)
+                    .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
+                    .setKeySize(MasterKey.DEFAULT_AES_GCM_MASTER_KEY_SIZE)
+                    .build()
```

Comment:
Is this following a recommended setup?
Do you have a reference link?

Reply:
Yes. Below are my references.
* https://developer.android.com/reference/androidx/security/crypto/EncryptedSharedPreferences
* https://devmainapps.blogspot.com/2020/06/android-masterkeys-deprecated-how-to.html
* https://stackoverflow.com/questions/62498977/how-to-create-masterkey-after-masterkeys-deprecated-in-android